### PR TITLE
py/objint: from_bytes: Make byteorder param optional again.

### PR DESCRIPTION
**Pull request by pfalcon**
It used to be optional for MicroPython, but was made required in 9d787de2a1.
However, many modules in micropython-lib still don't have it, which requires
extensive patching, require import sys module where it wasb't before, and
overall make memory requirements bigger and bytecode slower.

### CODE REVIEW

**Comment by pfalcon** [comment]
Commit message says it all: I made the mentioned patch, now I'd like to unmake it.

File: py/objint.c [hunk 1]
```diff
@@ -401,15 +401,23 @@ mp_obj_t mp_obj_int_binary_op_extra_cases(mp_binary_op_t op, mp_obj_t lhs_in, mp
...
 
     // get the buffer info
     mp_buffer_info_t bufinfo;
     mp_get_buffer_raise(args[1], &bufinfo, MP_BUFFER_READ);
 
+    #if MP_ENDIANNESS_LITTLE
+    bool big_endian = false;
+    #else
+    bool big_endian = true;
+    #endif
```
Comment (dpgeorge) [inline py/objint.c]: How about just `bool big_endian = MP_ENDIANNESS_BIG`?

File: py/objint.c [hunk 2]
```diff
@@ -420,26 +428,33 @@ STATIC mp_obj_t int_from_bytes(size_t n_args, const mp_obj_t *args) {
...
     if (len < 0) {
         mp_raise_ValueError(NULL);
     }
-    bool big_endian = args[2] != MP_OBJ_NEW_QSTR(MP_QSTR_little);
+
+    #if MP_ENDIANNESS_LITTLE
+    bool big_endian = false;
+    #else
+    bool big_endian = true;
+    #endif
```
Comment (dpgeorge) [inline py/objint.c]: Could be `bool big_endian = MP_ENDIANNESS_BIG`?

**Review by dpgeorge — COMMENTED** [review]
Well, it looks ok, and definitely makes sense to default to native byte ordering for these pair of functions (since if you pack and unpack on the same machine then you don't care the ordering).  Code size increase is minimal.  The only downside is that users get caught writing uPy-specfic code and forget to add the byte-order parameter, then it doesn't work on CPython.

Please write a test for the case when the arg is missing.

**Comment by pfalcon** [comment]
> The only downside is that users get caught writing uPy-specfic code and forget to add the byte-order parameter, then it doesn't work on CPython.

Yes, that's why submit it for review. Well, docs aren't going to describe that byteorder is optional, then the only reference would be CPython docs, which say it's required. Then the only way to discover it is randomly via REPL, but being undocumented and thus not having a well-defined behavior hopefully should deter its careless usage. So, it's really intended for uPy-only modules like micropython-lib, and by folks who know what they do ;-).

I'll make the changes you suggest, and please give it 2nd thought in the meantime, IMHO it's worth to have such shortcut.

**Comment by dpgeorge** [comment]
> IMHO it's worth to have such shortcut.

Please can you point to a few places in micropython-lib that really benefit from this shortcut.

**Comment by pfalcon** [comment]
Well, I'm not sure about "*really* benefit", but there were couple of fixes for that recently (10 months after change to uPy was introduced): https://github.com/micropython/micropython-lib/commit/20d9cdbd5f2c4de2a6bfa7d3ade8a5d74dfd8072 , https://github.com/micropython/micropython-lib/commit/2d4bc6997590724fe782f3ab8ab6b528dd78ed3b . I looked into improving micropython-lib regression testing, and saw more cases, e.g. multiprocessing module. So, they all need to be converted consistently, but that leads to a question of why the byteorder param was skipped in the first place, hence this patch.

Increase in the binary size if 28 bytes which is a little more than I'd expected. And by my own logic, anything which can be done in Python, ~~can~~ should be done there. So, I can't decide on this case, need 2nd opinion.

**Comment by dpgeorge** [comment]
> Well, I'm not sure about "really benefit", but there were couple of fixes for that recently ...
> ... and saw more cases, e.g. multiprocessing module

If they are the only cases (and grepping in micropython-lib shows they are) then I'd say it's not worth this patch.  In particular the multiproccessing module can use either little or big (and doesn't need to know the native endian) because it just writes the data to itself.  So it doesn't need to import sys.

In that case, -1 for this PR.

**Comment by pfalcon** [comment]
Indeed, those are the only cases. Closing then, until the need for such super-optimization comes.

**Pull request closed without merging**